### PR TITLE
layers: Remove unassigned image barrier aspect vuid

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Replaces #6906

### CODE REVIEW

**Comment by ziga-lunarg** [comment]
I think this needs a spec clarification, one of the tests crashes on AMD if we remove this unassigned VUID.

File: layers/state_tracker/image_state.cpp [hunk 1]
```diff
@@ -83,6 +83,21 @@ VkImageSubresourceRange NormalizeSubresourceRange(const VkImageCreateInfo &image
     return NormalizeSubresourceRange(image_create_info, subres_range);
 }
 
+VkImageSubresourceRange NormalizeSubresourceRangeWithAspect(const VkImageCreateInfo &image_create_info,
```
[the remark this answers is not in the document]
Comment (ziga-lunarg) [reply layers/state_tracker/image_state.cpp:86]: I've tried that in PR 6906, but that would remove some valid validation as well.

[the remark this answers is not in the document]
Comment (artem-lunarg) [reply layers/state_tracker/image_state.cpp:86]: For me it looks like equivalent code transformation, but yeah, I don't know much about context here.

File: tests/unit/sync_object.cpp [hunk 2]
```diff
@@ -397,7 +396,7 @@ TEST_F(NegativeSyncObject, Barriers) {
     // Having anything other than DEPTH and STENCIL is an error
     conc_test.image_barrier_.subresourceRange.aspectMask =
         VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT | VK_IMAGE_ASPECT_COLOR_BIT;
-    conc_test("UNASSIGNED-CoreValidation-DrawState-InvalidImageAspect");
+    conc_test("");
```
Comment (spencer-lunarg) [inline tests/unit/sync_object.cpp:399]: is this a positive test then?

File: tests/unit/sync_object.cpp [hunk 3]
```diff
@@ -1214,7 +1206,7 @@ TEST_F(NegativeSyncObject, DepthStencilImageNonSeparate) {
 
     // Not having DEPTH or STENCIL set is an error
     conc_test.image_barrier_.subresourceRange.aspectMask = VK_IMAGE_ASPECT_METADATA_BIT;
-    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "UNASSIGNED-CoreValidation-DrawState-InvalidImageAspect");
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "");
```
Comment (spencer-lunarg) [inline tests/unit/sync_object.cpp:1209]: can't we just remove the whole line now? I thought it was there as an extra VU

**Comment by spencer-lunarg** [comment]
This was discussed in https://gitlab.khronos.org/vulkan/vulkan/-/issues/3814

We will be adding a VU to ban the case of having an unused aspect mask for depth/stencil, so there is no need to "normalize" it 

I have also made other strides towards fixing this whole mess so this PR is even more out of date, but still being used as a reference

**Pull request closed without merging**